src/OpenApi/Schema.hs

```diff
@@ -195,15 +195,20 @@ newtype Enum = Enum [JSON.Value]
   deriving newtype (JSON.FromJSON, JSON.ToJSON)
   deriving stock   (Eq, Show)
 
-data Format = UnixTime
-  deriving stock (Eq, GHC.Bounded, GHC.Enum, Show)
+data Format
+  = CustomFormat Text  -- Non standard formats, explicitly allowed by OAS.
+  | UnixTime
+  deriving stock (Eq, Show)
 
 instance ToText Format where
   toText = \case
-    UnixTime -> "unix-time"
+    UnixTime            -> "unix-time"
+    CustomFormat format -> format
```

Comment:
Should these be sorted?

Reply:
yes, but we wanted to avoid question style unless its an actual question.

Reply:
There may be a semantic reason to leave it as-is, to correspond to the `FromJSON` instance, which is why I asked.

Reply:
ahh okay, no it can be sorted alphabetically as the pattern matches are unique individually. The order of patern matches only matters of one can be shadowed by the other. Which is not the case here was we are matching individual constructors that are mutually exclusive.